File: 백준/Silver/10773. 제로/제로.swift
var array = [Int]()
let K = Int(readLine()!)!
for _ in 1...K {
    let input = Int(readLine()!)!
    switch input {
    case 0:
        array.removeLast()
    default:
        array.append(input)
    }
}
print(array.reduce(0, +))

File: 백준/Silver/10773. 제로/제로.py
import sys

input = sys.stdin.readline
k = int(input())
arr = []
for _ in range(k):
    n = int(input())
    if n == 0 and arr:
        arr.pop()
    else:
        arr.append(n)
sys.stdout.write(str(sum(arr)) + "\n")